docker/gcp_pudl_etl.sh

```diff
@@ -99,29 +99,43 @@ function notify_slack() {
     send_slack_msg "$message"
 }
 
+function update_nightly_branch() {
+    git config --unset http.https://github.com/.extraheader
+    git config user.email "pudl@catalyst.coop"
+    git config user.name "pudlbot"
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
+    echo "WTAF: Updating nightly branch to point at $NIGHTLY_TAG."
+    git checkout nightly
+    git branch --set-upstream-to=origin/nightly
+    git pull
+    git merge --ff-only "$NIGHTLY_TAG"
+    ETL_SUCCESS=${PIPESTATUS[0]}
+    git push
+}
```

Comment:
This feels more involved than it needs to be. Is there a more concise way to do the same thing?  However, when it runs it seems to have the desired effect.

Reply:
I struggled a bit when trying to ensure that the `nightly` branch was able to be fast-forwarded to the new `$NIGHTLY_TAG`. For testing, I pointed `nightly` at `main` and since this branch is coming off of `main` it works. But I don't feel like I really understand how this will work in production.

Reply:
If:
- `nightly-XXXX-XX` is always an ancestor of `nightly-XXXX-XX+1`
- `nightly` always points at the current `nightly-XXXX` or a (chronologically) earlier one

Then, `nightly` will always be able to be fast-forwarded to the most current `nightly-XXXX-XX`. This is pretty easy to guarantee if `nightly-XXXX-XX` is always a commit on `main` and we don't rewrite history on `main`.

But, right now, since there's the whole `main`/`dev` split, it's a little harder to reason about. Thinking out loud:

- `nightly-XXXX-XX` is always at the tip of `dev` when this action runs.
- we don't move the history of `dev` around
- `nightly` always points at current or earlier `nightly-XXXX`

So I think the `--ff-only` should work.

Reply:
The first time I tried running it with `nightly` pointing at a prior nightly build and it couldn't fast-forward but I don't understand why.  Then I moved `nightly` to `main` for testing.

Reply:
Now that I've played with this more, I think it's because we're testing on this `debug-nightly-deploy` branch, and if `nightly` is pointing at a commit that's not in the `debug-nightly-deploy` history, the `--ff-only` will fail.